v4/feature/trade/src/main/java/exchange/dydx/trading/feature/trade/margin/DydxAdjustMarginInputViewModel.kt

```diff
@@ -84,4 +95,38 @@ class DydxAdjustMarginInputViewModel @Inject constructor(
             },
         )
     }
+
+    private fun validate(
+        input: AdjustIsolatedMarginInput,
+        marketMaxLeverage: Double,
+        position: SubaccountPosition,
+    ): String? {
+        val amount = input.amount ?: return null
+        val summary = input.summary ?: return null
+        val freeCollateral = position.freeCollateral.current
+        val marginUsage = position.marginUsage.postOrder
+
+        when (input.type) {
+            Add -> {
+                if (amount.toDouble() >= summary.crossFreeCollateral!!) {
```

Comment:
potentially risky to force unwrap if abacus is piecewise updating its summary state